src/features/user/Widget/DonationLink/DonationLinkForm.tsx

```diff
@@ -0,0 +1,244 @@
+import ProjectsList from '../../../projects/screens/Projects';
+import {
+  MuiAutocomplete,
+  StyledAutoCompleteOption,
+} from '../../../common/InputTypes/AutoCompleteTheme';
+
+// TODOO - refactor code for reuse?
+const StyledForm = styled('form')((/* { theme } */) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: 24,
+  alignItems: 'flex-start',
+  '& .formButton': {
+    marginTop: 24,
+  },
+  '& .inputContainer': {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: 24,
+    width: '100%',
+  },
+}));
+
+interface DonationLinkFormProps {
+  projectsList: Project[] | null;
+}
+
+const DonationLinkForm = ({
+  projectsList,
+}: DonationLinkFormProps): ReactElement | null => {
+  const { user, contextLoaded } = useContext(UserPropsContext);
+  const [country, setCountry] = useState(
+    contextLoaded ? user.country : undefined
+  );
+  const [Languages, setLanguage] = useState({
+    langCode: 'en',
+    languageName: 'English',
+  });
+  const [donationUrl, setDonationUrl] = useState<string>('');
+  const { t, ready } = useTranslation(['donationLink', 'donate']);
+  const [localProject, setLocalProject] = useState<Project | null>(null);
+  const [isSupport, setIsSupport] = useState<boolean>(!user.isPrivate);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [isTesting, setIsTesting] = useState<boolean>(false);
+  const [isProjectSelected, setIsProjectSelected] = useState<boolean>(false);
+  // Load all projects
+
+  const handleChange = () => {
+    const link = isTesting
+      ? 'http://paydev.pp.eco/'
+      : process.env.NEXT_PUBLIC_DONATION_URL;
+    const url = `${link}?country=${country}&locale=${Languages.langCode}${
+      localProject == null
+        ? ''
+        : `&to=${localProject.name.split(/\W+/).join('-').toLowerCase()}`
+    }&tenant=${TENANT_ID}${isSupport ? `&s=${user.slug}` : ''}
+    `;
+    setDonationUrl(url);
+  };
+
+  const handleProjectChange = async (project: Project | null) => {
+    setLocalProject(project);
+    setIsProjectSelected(!isProjectSelected);
+  };
+
+  useEffect(() => {
+    handleChange();
+  }, [country, Languages, localProject, isSupport, isTesting]);
+
+  // const setTextCopiedClipboard = () => {
+  //   navigator.clipboard.writeText(donationUrl);
+  //   setIsCopied(!isCopied);
+  // };
+
+  if (ready) {
+    return (
+      <StyledForm>
+        <div className="inputContainer">
+          <div className={styles.singleFormGroup}>
+            <div className={styles.formHeader}>
+              {t('donationLink:countryLanguageTitle')}
+            </div>
+            <InlineFormDisplayGroup>
+              <AutoCompleteCountry
+                label={t('donationLink:labelCountry')}
+                name="country"
+                defaultValue={country}
+                onChange={setCountry}
+              />
+              <MuiAutocomplete
+                id={t('donationLink:labelLanguages')}
```

Comment:
1. As a standard practice, make sure all form inputs have a name parameter. 
2. id/name do not need to be translatable.